airgun/widgets.py

```diff
@@ -267,6 +272,63 @@ class SatVerticalNavigation(VerticalNavigation):
     CURRENTLY_SELECTED = './/li[contains(@class, "active")]/a/span'
 
 
+class SatFlashMessages(FlashMessages):
+    """Satellite version of Patternfly's alerts section. The only difference is
+    overridden ``messages`` property which returns :class:`SatFlashMessage`.
+
+    Example html representation::
+
+        <div class="toast-notifications-list-pf">
+            <div class="alert toast-pf alert-success alert-dismissable">
+                <button ... type="button" class="close close-default">
+                    <span ... class="pficon pficon-close"></span></button>
+                <span aria-hidden="true" class="pficon pficon-ok"></span>
+                <span><span>Sample message</span></span></div>
+            </div>
+        </div>
+
+    Locator example::
+
+        //div[@class="toast-notifications-list-pf"]
+
+    """
+
+    @property
+    def messages(self):
```

Comment:
Okay, everything is inherited from `Widget` which doesn't have this nice `all` method which is good to use here
I am thinking about that checking visibility should be solely purpose of `satflashmessage` class.
I would more like to see something similar to `if `SatFlashMessage`.exists?=>append`. Is it possible to make more optimal that algorithm?

Reply:
I haven't touched original algorithm, it's completely the same as in `FlashMessages`, just returns our `SatFlashMessage` instead of `FlashMessage`. Imo, ideally such improvements should be considered in `widgetsastic_patternfly` repo, satellite's implementation should be dropped as soon as BZ1566565 is resolved.